dev/src/dev/portal.clj

```diff
@@ -68,3 +71,20 @@
                 {:portal.viewer/default :portal.viewer/diff})
               (update (meta middleware-var) :ns #(.name %)))
     (send-log event)))
+
+(defmacro diff->
+  "Drop-in replacement for `->` that sends diffs to Portal at each stage."
+  [x & forms]
+  #_{:clj-kondo/ignore [:discouraged-var]}
+  (loop [x x, forms forms]
+    (if forms
+      (let [form (first forms)
+            threaded `(-> ((fn [before#]
```

Comment:
You don't need the outer threading macro here, it's just taking the one form.

Reply:
I do, if I want this to really be a drop-in `->`, because `->` also handles (1) propagating metadata and (2) wrapping bare values in lists.

Reply:
Isn't the inner `->` on line 83 taking care of that? I'm talking about the one on line 82, which will just short circuit to identity.

Reply:
I agree with @crisptrutski, the outer `->` seem to be a no-op.